pkg/kube/client.go

```diff
@@ -550,14 +550,44 @@ func perform(infos Result, fn ResourceActorFunc) error {
 		return ErrNoObjectsVisited
 	}
 
-	for _, info := range infos {
-		if err := fn(info); err != nil {
+	errs := make(chan error)
```

Comment:
Nit: You could save the use of an errors channel by making the call to `batchPerform` blocking and then just returning a slice of errors from `batchPerform`. It would probably be less bug prone in the end, although the tradeoff would be that you'd need a mutex on the error slice in the function